hub/api/dataset.py

```diff
@@ -60,13 +59,14 @@ def __init__(
         self.storage = generate_chain(
             base_storage, memory_cache_size_bytes, local_cache_size_bytes, path
         )
-        self.tensors: Dict[str, Tensor] = {}
 
+        self.tensors: Dict[str, Tensor] = {}
         if dataset_exists(self.storage):
-            for tensor_name in self.meta["tensors"]:
+            self.meta = DatasetMeta.load(self.storage)
```

Comment:
`load` and `create` methods for the meta classes to me makes a lot of sense. it feels clearer to understand and read what's going on

Reply:
@McCrearyD I still prefer the initializer `DatasetMeta(self.storage)` instead of the explicit method `DatasetMeta.load(self.storage)`. They really mean the same thing, and the former is canonically correct.

I'm okay with a separate `.create` if it's sufficiently different from the initializer.